### DISCUSSION TITLE
`intermediates_dir` option for the `quarto render` command?

### DISCUSSION THREAD

**Discussion by eitsupi** [quarto-dev/quarto-cli#731] [Q&A] [answered]
Is it possible to change the name of the directory that contains the temporary files generated during rendering as in `rmarkdown::render()` function's `intermediates_dir` option.

When parallelizing a task that creates multiple files from a template by parameters, rendering will fail if the temp directory cannot be changed.
My use case is rocker-org/rocker-versioned2#322

**Comment by jjallaire** [comment, marked as answer]
We don't currently have this (and may not add it soon b/c there are so many codepaths that deal w/ paths and have an implicit dependency on the current behavior). For now I'd suggest staging a copy of the project dir in temp dir(s) for parallel renders.

**Reply by eitsupi** [reply]
Thank you for your quick answer!

**Comment by TinasheMTapera** [comment]
I know this thread is old, but the functionality is a high priority blocker for me, so some back and forth debugging and LLM bashing gave me a solution that seems to be working. I've asked the LLM to summarize+SEO the solution so that if anybody else is googling, hopefully it will be handy until the official team implements something:

> ChatGPT 5.5 incoming...

> We recently ran into this exact issue while parallelizing parameterized Quarto renders from an R `targets` pipeline.
> 
> What seems to work for us is a lighter-weight version of staging the whole project: copy the source `.qmd` to a uniquely named temporary `.qmd` in the same project directory, render that copy with its own parameters/output name, then immediately delete the temporary source file.
> 
> The important part is that each parallel render gets a unique source basename. Otherwise concurrent renders of the same `.qmd` appear to collide over Quarto-generated intermediates such as the `.quarto_ipynb` file and related cleanup.
> 
> Roughly:
> 
> ```r
> temp_qmd <- tempfile(
>   pattern = "download_era5_",
>   tmpdir = dirname(source_qmd),
>   fileext = ".qmd"
> )
> 
> file.copy(source_qmd, temp_qmd, overwrite = TRUE)
> 
> on.exit(
>   unlink(temp_qmd, force = TRUE),
>   add = TRUE
> )
> 
> quarto::quarto_render(
>   temp_qmd,
>   execute_params = params,
>   execute_dir = project_root,
>   output_file = output_file,
>   output_format = "gfm"
> )
> ```
> 
> Keeping the temporary copy beside the original has been useful because it remains in the same Quarto project and inherits the same `_quarto.yml`, extensions, etc., while still giving each render a separate intermediate-file namespace.
> 
> We have confirmed this under real parallel execution with multiple concurrent R/Quarto/Python processes. We do still occasionally see non-fatal cleanup warnings from Quarto’s session temp directories, but the rendered outputs complete successfully. The source-copy approach has been the key workaround for the actual concurrent intermediate-file collisions.
> 
> This also seems consistent with later workarounds discussed elsewhere in Quarto/tarchetypes: unique temporary source filenames appear to be enough in many cases, without needing to duplicate the entire project directory.
